pkg/network/ovn_kubernetes.go

```diff
@@ -2303,6 +2488,58 @@ func isInterConnectEnabledOnMasterDaemonset(ds *appsv1.DaemonSet) bool {
 	return isInterConnectEnabledOnDaemonset(ds, util.OVN_MASTER)
 }
 
+func isOVNIPsecNotActive(ds *appsv1.DaemonSet) bool {
+	// If no daemonset, then return false.
+	if ds == nil {
+		return false
+	}
+	// When observed generation doesn't match with spec generation or in progressing state
+	// then return false as we are not sure about IPsec state.
+	if ds.Generation != ds.Status.ObservedGeneration || daemonSetProgressing(ds, true) {
+		return false
+	}
+	annotations := ds.GetAnnotations()
+	// If OVN daemonset is set with IPsecEnableAnnotation, then return false.
+	if annotations[names.IPsecEnableAnnotation] != "" {
+		return false
+	}
+	// If IPsec is running with older version and ipsec=true is found from OVN daemonset, then return true.
+	if annotations["release.openshift.io/version"] != os.Getenv("RELEASE_VERSION") &&
```

Comment:
This should be `annotations["release.openshift.io/version"] < 4.15`

Use `isVersionGreaterThanOrEqualTo` for the comparison